src/Propel/Generator/Reverse/MysqlSchemaParser.php

```diff
@@ -114,6 +114,17 @@ public function parse(Database $database, array $additionalTables = [])
             $this->addPrimaryKey($table);
 
             $this->addTableVendorInfo($table);
+            
+         //Check if the table is the crosstab of a many-to-many relation
```

Comment:
Method is increasing in complexity with each new detail added, it would be great if someone could jump in and break up the jobs done in this method.

Reply:
Something with the indentation is awkward in this patch.